hoomd/mpcd/DryingDropletStreamingMethod.h

```diff
@@ -0,0 +1,295 @@
+
+// Maintainer: mphoward
+
+/*!
+ * \file mpcd/DryingDropletStreaming.h
+ * \brief Declaration of mpcd::DryingDropletStreaming
+ */
+
+#ifndef MPCD_DRYING_DROPLET_STREAMING_METHOD_H_
+#define MPCD_DRYING_DROPLET_STREAMING_METHOD_H_
+
+#ifdef NVCC
+#error This header cannot be compiled by nvcc
+#endif
+
+#include "ConfinedStreamingMethod.h"
+#include "hoomd/extern/pybind/include/pybind11/pybind11.h"
+#include "hoomd/Variant.h"
+#include "BoundaryCondition.h"
+#include "hoomd/RNGIdentifiers.h"
+#include "hoomd/RandomNumbers.h"
+#include <algorithm>
+
+namespace mpcd
+{
+//! MPCD DryingDropletStreamingMethod
+/*!
+ * This method implements the base version of ballistic propagation of MPCD
+ * particles in moving Spherical geometry.
+ *
+ * The integration scheme is essentially Verlet with specular reflections.First SphereGeometry radius and Velocity is updated then
+ * the particle is streamed forward over the time interval. If it moves outside the Geometry, it is placed back
+ * on the boundary and particle velocity is updated according to the boundary conditions.Streaming then continues and
+ * place the particle inside SphereGeometry. And particle which went outside or collided with geometry was marked.
+ * Right amount of marked particles are then evaporated.
+ *
+ * To facilitate this, ConfinedStreamingMethod must flag the particles which were bounced back from surface.
+ */
+
+class PYBIND11_EXPORT DryingDropletStreamingMethod : public mpcd::ConfinedStreamingMethod<mpcd::detail::SphereGeometry>
+    {
+    public:
+        //! Constructor
+        /*!
+         * \param sysdata MPCD system data
+         * \param cur_timestep Current system timestep
+         * \param period Number of timesteps between collisions
+         * \param phase Phase shift for periodic updates
+         * \param R is the radius of sphere
+         * \param bc is boundary conditions(slip or no-slip)
+         */
+        DryingDropletStreamingMethod(std::shared_ptr<mpcd::SystemData> sysdata,
+                                    unsigned int cur_timestep,
+                                    unsigned int period,
+                                    int phase, std::shared_ptr<::Variant> R, const Scalar density, unsigned int seed, mpcd::detail::boundary bc)
+        : mpcd::ConfinedStreamingMethod<mpcd::detail::SphereGeometry>(sysdata, cur_timestep, period, phase, std::shared_ptr<mpcd::detail::SphereGeometry>()),
+          m_R(R), m_density(density), m_seed(seed), m_picks(m_exec_conf), m_bc(bc)
+          {}
+
+        //! Implementation of the streaming rule
+        virtual void stream(unsigned int timestep);
+
+    protected:
+        std::shared_ptr<::Variant> m_R;      //!< Radius of Sphere
+        const Scalar m_density;              //!< Solvent density
+        unsigned int m_seed;                 //!< Seed to evaporator pseudo-random number generator
+        const mpcd::detail::boundary m_bc;   //!< Boundary condition
```

Comment:
Making these const means they can never be changed, even if you add a setter. I would probably not do that:
```suggestion
        mpcd::detail::boundary m_bc;   //!< Boundary condition
        Scalar m_density;              //!< Solvent density
        unsigned int m_seed;                 //!< Seed to evaporator pseudo-random number generator
```
(Also reorders per suggestion in constructor. Note you may need to change the order of the initialization list in the constructor to avoid a compiler error after making this move.)